packages/ui/utils/abbreviator.tsx

```diff
@@ -0,0 +1,15 @@
+import { Text } from '@chakra-ui/react';
+import { ReactNode } from 'react';
+
+import { PopoverTooltip } from '@ui/components/shared/PopoverTooltip';
+
+export function abbreviator(str: string | undefined, len: number): string | undefined | ReactNode {
+  if (str === undefined) return undefined;
+  return str?.length <= len ? (
+    str
+  ) : (
+    <PopoverTooltip placement="top-start" body={str}>
```

Comment:
Interesting to insert a popover around the abbreviated text to show the original string.
I would be a bit cautious if it really works as expected in every part of the app, as I don't know the inner workings of the Popover at all.